opendrift/readers/reader_netCDF_CF_unstructured.py

```diff
@@ -284,9 +285,12 @@ def get_variables(self,
                 sigmas = self.__nearest_node_sigma__(dvar, nodes, z)
                 assert len(sigmas) == len(z)
 
-                # for some unfathomable reason netCDF4 uses different fancy-indexing than numpy.
-                indx_t = np.repeat(indx_nearest, len(sigmas))
-                variables[var] = np.array([ dvar[ti, si, ni] for ti, si, ni in zip(indx_t, sigmas, nodes) ])
+                # Reading the smallest block covering the actual data
```

Comment:
Yes, can be moved out. But need to communicate whether face or node. Better to do so explicitly than with checking dimension. Maybe send siglay or siglay_center as argument.

Reply:
Hm, not face or node - but siglay or siglevel.